framework/PostProcessorFunctions/TypicalHistoryFromHistorySet.py

```diff
@@ -51,115 +69,168 @@ def run(self,inputDic):
-
-      return outputDic
+
+    #get actual data
+    inputDict = inputDic[0]['data']
+    #identify features
+    self.features = inputDict['output'][inputDict['output'].keys()[0]].keys()
+    #don't keep the pivot parameter in the feature space
+    if self.pivotParameter in self.features:
+      self.features.remove(self.pivotParameter)
+
+    #if output length (size of desired output history) not set, set it now
+    if self.outputLen is None:
+      self.outputLen = np.asarray(inputDict['output'][inputDict['output'].keys()[0]][self.pivotParameter])[-1]
+
+    # task: reshape the data into histories with the size of the output I'm looking for
+    #data dictionaries have form {historyNumber:{VarName:[data], VarName:[data]}}
+    reshapedData = {}
+    #keyNewH = 0
+    newHistoryCounter = 0 #new history tracking labels
+    for historyNumber in inputDict['output'].keys():
+      #array of the pivot values provided in the history
+      pivotValues = np.asarray(inputDict['output'][historyNumber][self.pivotParameter])
+      #if the desired output pivot value length is (equal to or) longer than the provided history ...
+      #   -> (i.e. I have a year and I want output of a year)
+      if self.outputLen >= pivotValues[-1]:
+        #don't change the shape of this history; it's fine as is
+        reshapedData[newHistoryCounter] = inputDict['output'][historyNumber]
+        newHistoryCounter += 1
+      #if the provided history is longer than the requested output period
+      #   -> (i.e., I have a year of data and I only want output of 1 year)
+      else:
+        #reshape the history into multiple histories to use
+        startPivot = 0
+        endPivot = self.outputLen
+        # until you find the last observed pivot point...
+        while endPivot <= pivotValues[-1]:
+          #create a storage place for each new usable history
+          reshapedData[newHistoryCounter] = {}
+          # acceptable is if the pivot value is greater than start and less than end
+          extractCondition = np.logical_and(pivotValues>=startPivot, pivotValues<=endPivot)
+          # extract out the acceptable parts from the pivotValues, and reset the base pivot point to 0
+          reshapedData[newHistoryCounter][self.pivotParameter] = np.extract(extractCondition, pivotValues)-startPivot
+          # for each feature...
+          for feature in self.features:
+            # extract applicable information from the feature set
+            reshapedData[newHistoryCounter][feature] = np.extract(extractCondition, inputDict['output'][historyNumber][feature])
+          #increment history counter
+          newHistoryCounter += 1
+          #update new start/end points for grabbing the next history
+          startPivot = endPivot
+          endPivot += self.outputLen
+
+    inputDict['output'] = reshapedData
+    self.numHistory = len(inputDict['output'].keys()) #should be same as newHistoryCounter - 1, if that's faster
```

Comment:
This doesn't need to be fixed, but sometimes comments like this make good warnings (as in if newHistoryCounter - 1 != len(...) raiseAWarning(...) )